### DISCUSSION TITLE
Devtool test Failing

### DISCUSSION THREAD

**Discussion by Ssunoo2** [sailuh/kaiaulu#263] [Q&A] [answered]
When running devtools::test(), I notice that a good number of tests fail but also notice in the error messages that many of the paths contain third party keywords such as "perceval". This leads to the question: are all of the third party tools required to be installed for the devtools tests to pass. 

Further, it is noted that the third party apps should be installed if the current notebooks use them as a dependency. Is there an easy way to find which third party apps are required for particular notebooks? Would it be a better idea to just install all third party apps anyway and not have to worry about it?

**Comment by carlosparadis** [comment, marked as answer]
> When running devtools::test(), I notice that a good number of tests fail but also notice in the error messages that many of the paths contain third party keywords such as "perceval". This leads to the question: are all of the third party tools required to be installed for the devtools tests to pass. 

Not all of them! Mostly unit tests that assess if Perceval is installed, or test parse_gitlog(), parse_mbox() are the ones that will fail. Other unit tests will load on the fly a data.table to test other functions. The unit tests for Perceval were moreso a warning to remind users that some of the functionality may be unavailable. Perhaps the message has to be updated to avoid confusion.

> Further, it is noted that the third party apps should be installed if the current notebooks use them as a dependency. Is there an easy way to find which third party apps are required for particular notebooks? Would it be a better idea to just install all third party apps anyway and not have to worry about it? 

Yes, you can see at the start of the Notebook which tools.yml tool is being loaded. I suppose a docker environment could be created, but that introduces another overhead. The third-party tools Kaiaulu depends on are relatively painless to set up: a pip install or download a binary. I guess Kaiaulu as a tool was created to facilitate, but not fully automate, the analysis process (since Software data tends to be messy and there is no silver bullet default). Generally, if you cannot set up the third-party tool yourself, that may already be a bad sign. You will also be unfamiliar with what the tool is doing and likely generate misleading analysis. Perhaps when the CLI comes around (which is part of the third milestone for this semester), this can be refined. 

Thanks for the insightful questions!
